### DISCUSSION TITLE
New osu mod `Keep in Scope`

### DISCUSSION THREAD

**Discussion by cdwcgt** [ppy/osu#28757] [Ideas]
this is a very easy impelment by using 
https://github.com/ppy/osu/pull/18626

let you fail when your cursor is outside of playfield border (which will affected by CS)

https://github.com/ppy/osu/assets/34775378/19d7bd70-f03f-4748-a8f7-2a8a614ab52d

Some sliders may eventually exceed the scope of the playfile border but there shouldn't be any problem?

Not sure if it is suitable as a mod for the official version, using this mod may make it impossible to play some aspire maps

**Comment by felipemarins** [comment]
While I think this can be easily cheesed by changing the tablet display area, I really think this can be a fun mod to play when using it correctly. And it seems like you already implemented it, so, why not?

**Comment by Avelia-tr** [comment]
I don't see what this mods add in current form. how likely is a player going out of those border in gameplay ? basically never. it could be made better if the border was in place where the cursor is meant to be. Currently this is just a fast retry button or frustrating way to die in breaks neither of wich are `needed`

How to fix that ? 
maybe make the border smaller depending on the effective play area ? it would still be not that good of a mod but it would do a bit more than nothing.

maybe make it so it consider only 2 hitobject to limit the box ? this could make the player require a more direct path for jump still it doesn't really affect gameply that much

**Reply by cdwcgt** [reply]
Yes, this is also a concern of mine, so I posted it on the discussion to seek some opinions.

In this way, there are many things that can be expanded. You can set how to determine the moving range, and even display the calculated movable range.
